base/server/python/pki/server/pkidestroy.py

```diff
@@ -202,14 +204,19 @@ def main(argv):
     parser.validate()
     parser.init_config(pki_instance_name=config.pki_deployed_instance_name)
 
-    log_dir = config.PKI_DEPLOYMENT_LOG_ROOT
-    log_name = "pki" + "-" +\
-               deployer.subsystem_name.lower() +\
-               "-" + "destroy" + "." +\
-               deployer.log_timestamp + "." + "log"
-    print('Uninstallation log: %s/%s' % (log_dir, log_name))
+    instance = pki.server.instance.PKIInstance(
+        parser.mdict['pki_instance_name'],
+        user=parser.mdict['pki_user'],
+        group=parser.mdict['pki_group'])
 
-    pkilogging.enable_pki_logger(log_dir, log_name, 'pkidestroy')
+    log_dir = os.path.join(instance.log_dir, deployer.subsystem_name.lower())
+    log_file = os.path.join(log_dir, 'uninstall.log')
+    print('Uninstallation log: %s' % log_file)
+
+    # create log file with the right permissions
+    instance.touch(log_file)
```

Comment:
Concern **2**:

 If I run plain `pkidestroy` without removing logs and respawn same subsystem with same instance (eg: pki-tomcat), this will fail, right?

Reply:
No, by default `Path.touch()` will allow existing file just like on Linux:
https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch

There was actually a problem creating the log dir if it already exists, but I've fixed that in the latest patch. Thanks for bringing up this test scenario!

When the instance is installed/removed again, pkispawn/pkidestroy will continue using the same file, appending the new log messages at the bottom just like any other log files. Also note that each log message inside the file has a timestamp, so the timestamp in the filename is not necessary and will clutter up the folder.